Packs/Base/Scripts/DrawRelatedIncidentsCanvas/DrawRelatedIncidentsCanvas.py

```diff
@@ -0,0 +1,121 @@
+    n_indicators = len(indicator_ids)
+    n_indicators_lines = int(n_indicators / (len(incident_ids) + 1)) + 1
+    indicators_per_line = math.ceil(n_indicators / n_indicators_lines)
+    for line in range(n_indicators_lines):
+        layer = line - int(n_indicators_lines / 2) if line < int(n_indicators_lines / 2) else line - int(
+            n_indicators_lines / 2) + 1
+        G.add_nodes_from(graph_indicators[indicators_per_line * line:indicators_per_line * (line + 1)], layer=layer)
+
+    for connection in connections:
+        src = "%s-%s" % (connection['srcEntityType'], connection['srcEntityId'])
+        dest = "%s-%s" % (connection['targetEntityType'], connection['targetEntityId'])
+        G.add_edge(src, dest)
+    pos = layout_to_functions.get(LAYOUT)(G)  # type: ignore
+    layout = {}
+    for k, v in pos.items():
+        _type = k.split("-")[0]
+        layout[k] = convert_position(k, _type, v)
+
+    if LAYOUT == 'shell':
+        layout["incident-" + current_incident_id]['position']["x"] = 0
+        layout["incident-" + current_incident_id]['position']["y"] = 0
+    return layout
+
+
+def get_incident_to_incident_connection(inc1, inc2):
+    return {"srcEntityType": "incident", "srcEntityId": inc1, "targetEntityType": "incident", "targetEntityId": inc2}
+
+
+def get_incident_to_indicator_connection(inc, ioc):
+    return {"srcEntityType": "incident", "srcEntityId": inc, "targetEntityType": "indicator", "targetEntityId": ioc}
+
+
+def generate_canvas(current_incident_id, incident_ids, indicators):
+    indicator_ids = list(map(lambda x: x['id'], indicators))
+
+    incident_nodes = list(map(lambda x: {'type': 'incident', 'id': x}, incident_ids))
+    indicators_nodes = list(map(lambda x: {'type': 'indicator', 'id': x}, indicator_ids))
+    nodes = incident_nodes + indicators_nodes
+
+    incident_to_other_incidents = []
+    for incident_id in incident_ids:
+        if incident_id != current_incident_id:
+            incident_to_other_incidents.append(get_incident_to_incident_connection(current_incident_id, incident_id))
+
+    incident_to_indicator_connections = []
+    for indicator in indicators:
+        for investigation_id in indicator.get('investigationIDs', []):
+            if investigation_id in incident_ids + [current_incident_id]:
+                incident_to_indicator_connections.append(
+                    get_incident_to_indicator_connection(investigation_id, indicator['id']))
+
+    connections = incident_to_other_incidents + incident_to_indicator_connections
+    canvas = {
+        "incidentID": current_incident_id,
+        "version": -1,
+        'nodes': nodes,
+        'layout': generate_layout(current_incident_id, incident_ids, indicator_ids, connections)
+    }
+    return canvas, connections
+
+
+def main():
+    if not is_demisto_version_ge(MINIMUM_SERVER_VERSION):
+        return_error('This script is supported only from version {}.'.format(MINIMUM_SERVER_VERSION))
```

Comment:
@yaakovi can you make sure I used it correctly? thanks

Reply:
looks good